tests/integration/psync2.tcl

```diff
@@ -109,6 +110,25 @@ start_server {} {
         test "PSYNC2: --- CYCLE $cycle ---" {}
         incr cycle
 
+        # Build a lookup with the root master of each replica (head of the chain).
+        array set root_master {}
+        for {set j 0} {$j < 5} {incr j} {
+            set r $j
+
+            while {1} {
+                set r_master_port [status $R($r) master_port]
+
+                if {$r_master_port == ""} {
+                    set root_master($r) $r
```

Comment:
```suggestion
                    set root_master($j) $r
```

Reply:
A little embarrassing. Indeed a bug.
This line make sure that lookup always `self to self`
So the `wait_for_condition` part will always true. 
(So it didn't actually work)

I did debug and printed `root_master` before, (Actually saw this)
but I wasn’t clear enough to find out the `self to self` thing...

I apply the suggestion.  Now it look good to me..
Also please check my last commit 
(The git log is not pretty, it will be ok when we squash)